test/integration/supervisor_tls_spec_test.go

```diff
@@ -0,0 +1,266 @@
+		},
+		{
+			name: "should disallow certificate authority data source with empty value kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: invalid-oidc-idp-invalid-kind
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: ""
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "invalid-oidc-idp-invalid-kind",
+			expectedError:      `The OIDCIdentityProvider "invalid-oidc-idp-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should disallow certificate authority data source with invalid kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: invalid-oidc-idp-invalid-kind
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: sorcery
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "invalid-oidc-idp-invalid-kind",
+			expectedError:      `The OIDCIdentityProvider "invalid-oidc-idp-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "sorcery": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should create a custom resource passing all validations using a Secret source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: valid-oidc-idp-secret-kind1
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "valid-oidc-idp-secret-kind1",
+			expectedError:      "",
+		},
+		{
+			name: "should create a custom resource passing all validations using a ConfigMap source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: valid-oidc-idp-cm-kind1
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: ConfigMap
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
```

Comment:
The controller will really make HTTPS requests to `foo.bar.com` so it would be better to use the real configured OIDC issuer from the test environment for an integration test.

Reply:
This test validates the kubebuilder validations that are part of the spec. So these resources can be created in the `default` namespace and don't have to be created in the `pinniped-supervisor` namespace. Because the resources are created in the `default` namespace, they will not be reconciled by the controllers. So just in this test, it is OK to use dummy data without having to add complexity to generating the test data. I am going to mark this as resolved. Please feel free to unresolve.